### ISSUE TITLE
sklearn.model_selection.cross_val_score execution timeout

### ISSUE THREAD

**Issue by fent665** [googlecolab/colabtools#2410]
Running this cell of code inside of a colab notebook takes way longer than it should and eventually times out.
    
    from sklearn.model_selection import cross_val_score

    base_score = cross_val_score(knn_clf, X_train, y_train, cv=2, scoring="accuracy")

For reference, this same cell executed in 1m 20s on my 2C 4T laptop.

The code before the aforementioned cell, all of which runs perfectly fine:

    from sklearn.datasets import fetch_openml
    
    mnist = fetch_openml('mnist_784', version=1)
    X, y = mnist["data"], mnist["target"]
    X_train, y_train, X_test, y_test = X[:60000], y[:60000], X[60000:], y[60000:]
    
    from sklearn.neighbors import KNeighborsClassifier
    
    knn_clf = KNeighborsClassifier()
    knn_clf.fit(X_train, y_train)

**Comment by craigcitro** [comment]
Thanks for the report! It looks like this is due to our ancient `scikit-learn` version; I'll work on getting that updated.

**Comment by craigcitro** [comment]
To be clear: an immediate workaround is `!pip install --upgrade scikit-learn`.

**Comment by craigcitro** [comment]
This should be fixed. Thanks again for the report!
